### PR TITLE
Simplify logic and add test

### PR DESCRIPTION
**Pull request by helen-frank**
**Please provide a description of this PR:**

removeDNSConfig: Simplify logic and add test

### CODE REVIEW

File: istioctl/cmd/kubeuninject.go [hunk 1]
```diff
@@ -197,23 +197,10 @@ func removeDNSConfig(podDNSConfig *corev1.PodDNSConfig) {
...
-					podDNSConfig.Searches[index+1:]...)
-			} else {
-				podDNSConfig.Searches = podDNSConfig.Searches[:index]
-			}
-			// reset to 0
-			index = 0
-			l = len(podDNSConfig.Searches)
-		} else {
-			index++
+	for i := 0; i < len(podDNSConfig.Searches); i++ {
```
Comment (hanxiaop) [inline istioctl/cmd/kubeuninject.go:200]: Please run `make gen` to include any changed files

Comment (hanxiaop) [reply istioctl/cmd/kubeuninject.go:200]: The code LGTM

Comment (helen-frank) [reply istioctl/cmd/kubeuninject.go:200]: Use `make gen` , `git diff.` found that nothing has changed
Finally use git apply <(curl -sL "https://gcsweb.istio.io/gcs/istio-prow/pr-logs/pull/istio_istio/43901/gencheck_istio/1635356338655596544/artifacts/check-clean-repo-diff.patch")

Comment (hanxiaop) [reply istioctl/cmd/kubeuninject.go:200]: This is strange. The code you've written should not affect protoc, maybe try to add the code to the latest commit. I haven't encountered this issue today.

Comment (helen-frank) [reply istioctl/cmd/kubeuninject.go:200]: I tried to switch to the master branch using `git fetch --all && git rebase upstream/master && git switch feature/OptimizationremoveDNSConfig && git rebase upstream/master`, but nothing changes, and I tried `git push -f`

**Review by hanxiaop — APPROVED** [review]
LGTM, thanks

**Pull request merged**